vortex-file/src/read/stream.rs

```diff
@@ -50,22 +47,40 @@ impl<R: VortexReadAt> VortexFileArrayStream<R> {
         row_count: u64,
         row_mask: Option<RowMask>,
         dispatcher: Arc<IoDispatcher>,
-    ) -> Self {
+    ) -> VortexResult<Self> {
+        let mut reader_splits = BTreeSet::new();
```

Comment:
since we can construct all the readers eagerly now it's simplest to register all splits on creation